runtime/litmus/src/tests/xcm_parachain/mod.rs

```diff
@@ -112,64 +82,64 @@ fn test_xtokens_recognize_multilocation() {
 			Box::new(
 				(Parent, Parachain(2), Junction::AccountId32 { network: Any, id: BOB }).into()
 			),
-			800_000_000
+			UnitWeightCost::get() * 4
 		));
-	});
-
-	// TODO::This mimic does not finialized yet. Still need mimic XcmRouter
-	// ParaB::execute_with(|| {
-	// 	XcmpQueue::on_idle(0, u64::MAX);
-	// 	assert_eq!(Tokens::free_balance(0, &AccountId::from(BOB)), 999_200_000_000);
-	// })
-}
-
-#[test]
-fn test_pallet_xcm_recognize_multilocation() {
-	TestNet::reset();
-	Relay::execute_with(|| {
-		let _ = pallet_balances::Pallet::<relay::Runtime>::deposit_creating(
-			&para_a_account(),
-			1_000_000_000_000,
+		assert_eq!(
+			Balances::free_balance(&AccountId::from(ALICE)),
+			500_000_000_000_000_000 - 1_000_000_000_000
 		);
-		let _ = pallet_balances::Pallet::<relay::Runtime>::deposit_creating(
-			&para_b_account(),
-			1_000_000_000_000,
+		assert_eq!(
+			Balances::free_balance(&sibling_account(2)),
+			1_000_000_000_000 /* Notice this is interesting, as it suggest local preserve XCM
+			                   * fee belongs to remote chain, not local chain */
 		);
 	});
-	ParaA::execute_with(|| {
-		register_channel_info(1, 2);
-	});
-	ParaB::execute_with(|| {
-		register_channel_info(2, 1);
-	});
+
 	ParaB::execute_with(|| {
-		assert_ok!(AssetManager::register_foreign_asset_type(
-			RawOrigin::Root.into(),
-			CurrencyId::ParachainReserve(Box::new(
-				(
-					Parent,
-					Parachain(1),
-					PalletInstance(<Balances as PalletInfoAccess>::index() as u8)
-				)
-					.into()
-			)),
-			Default::default()
-		));
-		assert_ok!(AssetManager::set_asset_units_per_second(
-			RawOrigin::Root.into(),
-			1,
-			1_000_000_000_000
+		assert_eq!(
+			Tokens::free_balance(
+				1, // Asset_id=1. The first registered Token in Para B
+				&AccountId::from(BOB)
+			),
+			u128::from(1_000_000_000_000 - UnitWeightCost::get() * 4)
+		);
+
+		// Send ParaA token back to ParachainA's BOB
+		assert_ok!(XTokens::transfer(
+			Origin::signed(AccountId::from(BOB)),
+			CurrencyId::ParachainReserve(Box::new(para_native_token_multilocation(1))),
+			500_000_000_000,
+			Box::new(
+				(Parent, Parachain(1), Junction::AccountId32 { network: Any, id: BOB }).into()
+			),
+			UnitWeightCost::get() * 4
 		));
 	});
 
+	ParaA::execute_with(|| {
+		assert_eq!(
+			Balances::free_balance(&AccountId::from(BOB)),
+			u128::from(500_000_000_000 - UnitWeightCost::get() * 4)
+		);
+		assert_eq!(
+			Balances::free_balance(&sibling_account(2)),
+			500_000_000_000 /* When assets transfer back, the xcm fee is moved to
```

Comment:
`500_000_000_000/500_000_000_000` might not be the best testcase here, as you can't tell which one is subtractor which one is remainder => maybe 6/4 or 7/3

Reply:
Changed